### PR TITLE
Setuniform

### LINKED ISSUE #5512: Remove UV warning when setting textures with setUniform()
#### Most appropriate sub-area of p5.js?

- [ ] Accessibility (Web Accessibility)
- [ ] Build tools and processes
- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] DOM
- [ ] Events
- [ ] Friendly error system
- [ ] Image
- [ ] IO (Input/Output)
- [ ] Localization
- [ ] Math
- [ ] Unit Testing
- [ ] Typography
- [ ] Utilities
- [X] WebGL
- [ ] Other (specify if possible)

#### Details about the bug:

If you are using the vertex function and supplying your own UV's,  when using the `setUniform()` function to send a texture to a custom shader, a warning is triggered telling you: "You must first call texture() before using vertex() with image based u and v coordinates ". 

Since we're supplying the texture with `setUniform()`, we shouldn't be showing this warning.

Actually, I think there is a case to be made to remove this warning entirely. It could be that the user is writing a procedural shader, and not using textures, in which case, the warning isn't necessary at all.

The warning seems to go away when using textureMode(NORMAL);

- p5.js version: 1.4
- Web browser and version: Chrome 96
- Operating System: Mac OSX
- Steps to reproduce this:

1. Load a custom shader that samples a texture
2. Send the texture to the shader
3. Draw some geometry with beginShape(), vertex() ....
4. Observe the warning.

Here's a [small example showing the issue](https://editor.p5js.org/aferriss/sketches/Irz7B8DYy)

### PR DESCRIPTION
**Pull request by Gaurav-1306**
Resolves #5512

 Changes:

Added a elseif statement to make sure no error message is thrown when user defined texture is used.

 Screenshots of the change:

#### PR Checklist

- [x ] `npm run lint` passes
- [ x] [Inline documentation] is included / updated
- [ x] [Unit tests] are included / updated

[Inline documentation]: https://github.com/processing/p5.js/blob/main/contributor_docs/inline_documentation.md
[Unit tests]: https://github.com/processing/p5.js/tree/main/contributor_docs#unit-tests

### CODE REVIEW

**Comment by Gaurav-1306** [comment]
@davepagurek  can you look at the code and merge the commit?

File: src/webgl/p5.RendererGL.Immediate.js [hunk 1]
```diff
@@ -134,6 +134,12 @@ p5.RendererGL.prototype.vertex = function(x, y) {
         u /= this._tex.width;
         v /= this._tex.height;
       }
+    } if (
+      (this.userFillShader !== undefined ||
```
Comment (davepagurek) [inline src/webgl/p5.RendererGL.Immediate.js:138]: Just a minor style comment: I think we can take out the extra set of brackets around the if condition.

**Review by davepagurek — COMMENTED** [review]
Thanks for making this PR! I just have one very minor comment to clean up the code, and then this is good to go!

File: src/webgl/p5.RendererGL.Immediate.js (hunk 1 above)
Comment (Gaurav-1306) [reply src/webgl/p5.RendererGL.Immediate.js:138]: Good point. Let me make the changes!

**Review by davepagurek — APPROVED** [review]
Thanks @Gaurav-1306!

File: src/webgl/p5.RendererGL.Immediate.js [hunk 2]
```diff
@@ -134,6 +134,12 @@ p5.RendererGL.prototype.vertex = function(x, y) {
         u /= this._tex.width;
         v /= this._tex.height;
       }
+    } if (
```
Comment (davepagurek) [inline src/webgl/p5.RendererGL.Immediate.js:137]: Oh one more thing, should this be an else if?

```suggestion
    } else if (
```

**Review by davepagurek — CHANGES_REQUESTED** [review]

File: src/webgl/p5.RendererGL.Immediate.js (hunk 2 above)
Comment (Gaurav-1306) [reply src/webgl/p5.RendererGL.Immediate.js]: > Oh one more thing, should this be an else if?

okay cool, I thought they would do the name thing.

**Review by davepagurek — APPROVED** [review]

**Pull request merged**